py/persistentcode.c

```diff
@@ -72,6 +72,20 @@ typedef struct _bytecode_prelude_t {
 static int read_byte(mp_reader_t *reader);
 static size_t read_uint(mp_reader_t *reader);
 
+#if MICROPY_PERSISTENT_CODE_TRACK_FUN_DATA || MICROPY_PERSISTENT_CODE_TRACK_BSS_RODATA
+
+// An mp_obj_list_t that tracks native text/BSS/rodata to prevent the GC from reclaiming them.
+MP_REGISTER_ROOT_POINTER(mp_obj_t persistent_code_root_pointers);
+
+static void track_root_pointer(void *ptr) {
+    if (MP_STATE_PORT(persistent_code_root_pointers) == MP_OBJ_NULL) {
+        MP_STATE_PORT(persistent_code_root_pointers) = mp_obj_new_list(0, NULL);
+    }
+    mp_obj_list_append(MP_STATE_PORT(persistent_code_root_pointers), MP_OBJ_FROM_PTR(ptr));
```

Comment:
I'm guessing there's a good reason this doesn't work, but I'm curious what it is: What about extending `mp_raw_code_t` (possibly by creating a new wrapper struct that's optionally subbed in for `mp_raw_code_t *rc = mp_emit_glue_new_raw_code();`), and then append the code pointer to a field at the end of it? That way couldn't the raw code pointers stay valid until all of the raw code objects were GCed, but then be GCed themselves if the code falls out of scope?

Reply:
Got it, thanks.